### PR TITLE
ON-4787 # Changed constructors and class structure of some Job relate…

### PR DESCRIPTION
**Pull request by Zaxist**
Changed constructors and class structure of some Job related classes. 

Removed NewJob class.

Added documentation.

### CODE REVIEW

File: OneBlink.SDK.Tests/JobsClientTests.cs [hunk 1]
```diff
@@ -151,7 +166,12 @@ public async void throws_error_if_title_empty()
 
                 JobDetail jobDetail = new JobDetail("");
 
-                NewJob newJob = new NewJob(jobDetail, formId, "developers@oneblink.io");
+                //NewJob newJob = new NewJob(jobDetail, formId, "developers@oneblink.io");
```
Comment (mymattcarroll) [inline OneBlink.SDK.Tests/JobsClientTests.cs]: Ideally we don't have commented out code in our code bases. Please remove.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+### Changed 
+
+-   the properties and class structure of [`Job()`], [`JobDetail()`] and [`JobsSearchParameters()`]
+-   Constructors for each class can now used instantiate all relevant properties
+-   The [`Job()`] class is now passed to [`JobsClient.createJob()`] instead of [`NewJob()`]
+
+### Changed 
+
+-   The name of [`JobsClient.search()`] function to [`JobsClient.searchJobs()`]
+
```
Comment (mymattcarroll) [inline CHANGELOG.md]: As discussed, please remove.

File: OneBlink.SDK/models/Job.cs [hunk 3]
```diff
@@ -84,76 +98,173 @@ public string type
...
+        // public bool isSubmitted
+        // {
+        //     get; set;
+        // }
+
+        // public DateTime createdAt
+        // {
+        //     get; set;
+        // }
+    // }
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Job.cs]: Please remove.

File: OneBlink.SDK/models/Job.cs [hunk 4]
```diff
@@ -84,76 +98,173 @@ public string type
...
+        //     this.username = username;
+        // }
+
+        // public NewJob(JobDetail details, int formId, string username, string externalId)
+        // {
+        //     this.details = details;
+        //     this.formId = formId;
+        //     this.username = username;
+        //     this.externalId = externalId;
+        // }
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Job.cs]: Please remove.

File: OneBlink.SDK/models/Job.cs [hunk 5]
```diff
@@ -84,76 +98,173 @@ public string type
...
+//     }
 
-        public DateTime createdAt
-        {
-            get; set;
-        }
-    }
-}
\ No newline at end of file
+// }
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Job.cs]: Please remove.

File: docs/README.md [hunk 6]
```diff
@@ -6,3 +6,4 @@
 -   [PdfClient](./pdf-client.md)
 -   [TeamMembersClient](./team-members-client.md)
 -   [JobsClient](./jobs-client.md)
+-   [Models](./models.md)
```
Comment (mymattcarroll) [inline docs/README.md]: Could we please create a file for each class and maybe have a new sub-heading for `Models` and put all the Client classes under a sub-heading `Clients`

File: OneBlink.SDK.Tests/JobsClientTests.cs (hunk 1 above)
Comment (Zaxist) [reply OneBlink.SDK.Tests/JobsClientTests.cs]: Oops, forgot to remove it. Will do.

File: OneBlink.SDK/models/Job.cs [hunk 7]
```diff
@@ -84,76 +98,71 @@ public string type
...
         }
-    }
 
-    public class Job : NewJob
-    {
+        //INTERAL SET PROPERTIES, NO EXTERNAL SETTING FOR SETTING
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string id
         {
             get; set;
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Job.cs]: Should this one have `internal` too?

File: docs/models/job.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+### Example
+
+```c#
+OneBlink.SDK.Model.Job jobToCreate = new OneBlink.SDK.Model.Job(
+  username: "developer@oneblink.io",
+  formId: 4,
+  details: jobToCreateDetails,
+  externalId: "MyExternalId",
+  prefillFormDataId: "MyPrefillFormDataId"
```
Comment (mymattcarroll) [inline docs/models/job.md]: This should be in the extra properties, we dont want our customers setting this value.

File: docs/models/job.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+## `Job()`
+
+### Constructor
+
+| Parameter           | Required | Type                          | Description                               | Default Value |
+| ------------------- | -------- | ----------------------------- | ----------------------------------------- | ------------- |
+| `username`          | No       | `string`                      | Username associated with the job.         | null          |
+| `formId`            | No       | `int`                         | formId associated with the job.           | null          |
+| `details`           | No       | [`JobDetail`](<#JobDetail()>) | Instance of the job's details.            | null          |
```
Comment (mymattcarroll) [inline docs/models/job.md]: Should link to another file now.

File: OneBlink.SDK/models/Job.cs [hunk 10]
```diff
@@ -84,76 +98,71 @@ public string type
...
-
-        public int formId
+        public string username
         {
             get; set;
         }
 
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public string preFillFormDataId
+        public Nullable<int> formId
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Job.cs]: Why is this nullable?

Comment (Zaxist) [reply OneBlink.SDK/models/Job.cs]: Shouldn't be.

File: OneBlink.SDK/models/Job.cs (hunk 7 above)
Comment (Zaxist) [reply OneBlink.SDK/models/Job.cs]: yep

File: docs/models/job.md (hunk 8 above)
Comment (Zaxist) [reply docs/models/job.md]: fixed.

File: docs/models/job.md (hunk 9 above)
Comment (Zaxist) [reply docs/models/job.md]: fixed.

**Pull request merged**